contracts/strategies/mstable/IdleMStableStrategy.sol

```diff
@@ -0,0 +1,237 @@
+    IERC20Detailed public underlyingToken;
+
+    /* ------------Extra declarations ---------------- */
+    address public govToken;
+    IVault public vault;
+
+    address public idleCDO;
+    address[] public uniswapRouterPath;
+    IUniswapV2Router02 public uniswapV2Router02;
+
+    uint256 public lastIndexAmount;
+    uint256 public lastIndexedTime;
+
+    uint256 constant YEAR = 365 days;
+
+    constructor() {
+        token = address(1);
+    }
+
+    function initialize(
+        address _strategyToken,
+        address _underlyingToken,
+        address _vault,
+        address _idleCDO,
+        address _uniswapV2Router02,
+        address[] calldata routerPath,
+        address _owner
+    ) public initializer {
+        OwnableUpgradeable.__Ownable_init();
+        ReentrancyGuardUpgradeable.__ReentrancyGuard_init();
+
+        //----- // -------//
+        strategyToken = _strategyToken;
+        token = _underlyingToken;
+        underlyingToken = IERC20Detailed(token);
+        tokenDecimals = underlyingToken.decimals();
+        oneToken = 10**(tokenDecimals);
+        imUSD = ISavingsContractV2(_strategyToken);
+        vault = IVault(_vault);
+        govToken = vault.getRewardToken();
+        idleCDO = _idleCDO;
+
+        uniswapRouterPath = routerPath;
+        uniswapV2Router02 = IUniswapV2Router02(_uniswapV2Router02);
+
+        ERC20Upgradeable.__ERC20_init("Idle MStable Strategy Token", string(abi.encodePacked("idleMS", underlyingToken.symbol())));
+        lastIndexedTime = block.timestamp;
+        //------//-------//
+
+        transferOwnership(_owner);
+    }
+
+    // only claim gov token rewards
+    function redeemRewards() external onlyIdleCDO returns (uint256[] memory rewards) {
+        _claimGovernanceTokens(0, 0);
+        rewards = new uint256[](1);
+        rewards[0] = _swapGovTokenOnUniswapAndDepositToVault(0); // will redeem whatever possible reward is available
+    }
+
+    function redeemRewards(bytes calldata _extraData) external override onlyIdleCDO returns (uint256[] memory rewards) {
+        (uint256 minLiquidityTokenToReceive, uint256 startRound, uint256 endRound) = abi.decode(_extraData, (uint256, uint256, uint256));
+        _claimGovernanceTokens(startRound, endRound);
+        rewards = new uint256[](1);
+        rewards[0] = _swapGovTokenOnUniswapAndDepositToVault(minLiquidityTokenToReceive);
+    }
+
+    function pullStkAAVE() external override returns (uint256) {
+        return 0;
+    }
+
+    function price() public view override returns (uint256) {
+        return imUSD.exchangeRate();
+    }
+
+    function getRewardTokens() external view override returns (address[] memory) {
+        address[] memory govTokens;
+        govTokens[0] = govToken;
+        return govTokens;
+    }
+
+    function deposit(uint256 _amount) external override onlyIdleCDO returns (uint256 minted) {
+        require(_amount != 0, "Deposit amount should be greater than 0");
```

Comment:
If we follow the conventions, we should not revert when `_amount`  is zero.
```
https://github.com/idlehusbandry/idle-tranches/blob/master/contracts/strategies/idle/IdleStrategy.sol#L78

Reply:
Yes,
Changed